plugins/SliceInfoPlugin/SliceInfo.py

```diff
@@ -119,6 +119,7 @@ def _onWriteStarted(self, output_device):
             }
             for container in global_container_stack.getContainers():
                 container_id = container.getId()
+                container_serialized = ""
```

Comment:
The for loop starts (line 120). The first container is not serialisable, so you hit the `Except` in line 125, and consequently the `continue`, and you start over at line 120. In that loop, `container_serialized` is indeed undefined but you never hit the check against it.

For the second container, it tries again, but this time it's serialisable so line 124 successfully executes, `container_serialized` is defined, and at line 129, the `if` statement becomes `True` (as long as the resulting serialisation is not empty).

I think I'm missing something here then? It's not terribly important, of course...